File: api/javaspring/src/main/java/com/robinsonalexk/todospring/repositories/TodoRepository.java
package com.robinsonalexk.todospring.repositories;

import com.robinsonalexk.todospring.models.TodoItem;
import org.springframework.data.repository.CrudRepository;

import java.util.List;


public interface TodoRepository {

    TodoItem findById(int id);
    List<TodoItem> findByStatus(String status);
    List<TodoItem> findAll();
    void insert(TodoItem item);
    void update(TodoItem item);
    void delete(TodoItem item);

}

File: api/netcore/todo-netcore/todo-netcore/Repositories/TodoRepository.cs
using Microsoft.EntityFrameworkCore;
using TodoNETCore.Data;

namespace TodoNETCore.Repositories
{
    public class TodoRepository : ITodoRepository
    {
        private readonly TodoDbContext _context;

        public TodoRepository(TodoDbContext context) 
        { 
            _context = context; 
        }

        public async Task<TodoItem> FindById(long id)
        {
            return await _context.TodoItems.FirstAsync(x => x.Id == id);
        }

        public async Task<IEnumerable<TodoItem>> FindAll()
        {
            return await _context.TodoItems.ToListAsync();
        }

        public async Task<IEnumerable<TodoItem>> FindByStatus(string status)
        {
            return await _context.TodoItems.Where(x => x.Status == status).ToListAsync();
        }

        public void Insert(TodoItem item)
        {
            _context.TodoItems.Add(item);
            _context.SaveChanges();
        }

        public void Update(TodoItem item)
        {
            throw new NotImplementedException();
        }

        public void Delete(TodoItem item)
        {
            throw new NotImplementedException();
        }
    }
}
